src/api/torii-endpoints.md

```diff
@@ -0,0 +1,370 @@
+depending on the information found in the block.
+
+### Blocks Stream
+
+- **Protocol:** HTTP
+- **Protocol Upgrade:**`WebSocket
+- **Endpoint:** `/block/stream`
+
+#### Pipeline
+
+The client should send a [`BlockSubscriptionRequest`](/api/data-model-schema#blocksubscriptionrequest) to initiate
+communication after the WebSocket handshake. Then the server sends a [`BlockMessage`](/api/data-model-schema#blockmessage).
+Messages are encoded with the [Parity Scale Codec](#parity-scale-codec).
+
+#### Notes
+
+Via this endpoint, the client first provides the starting block number (i.e. height) in the subscription request. After
+sending the confirmation message, the server starts streaming all the blocks from the given block number up to the current
+block and continues to stream blocks as they are added to the blockchain.
+
+### Get Configuration
+
+::: warning
+
+TODO: Will change as part of the config RFC
+
+:::
+
+- **Protocol:** HTTP
+- **Method:** `GET`
+- **Endpoint:** `/configuration`
+
+There are 2 possible request bodies and according responses.
+
+#### Get Configuration Value as JSON
+
+- **Request Body:**
+  ```json
+  "Value"
+  ```
+- **Responses:**
+  - `200 OK`: Configuration value as JSON
+
+#### Get Documentation of a Field
+
+- **Request Body:**
+  ```json
+  {
+    "Docs": ["a", "b", "c"]
+  }
+  ```
+  where "a.b.c" is a path of the field (e.g. `sumeragi.block_time_ms`).
+- **Responses:**
+  - `200 OK`: Field was found and either doc or value is returned in json body.
+  - `404 Not Found`: Field wasn't found
+
+::: tip
+
+If the requested field has more fields inside of it, then all the documentation for its inner members is returned as
+well. Here is an example for getting a field `a.b.c`:
+
+**Example:** getting top-level documentation for `Torii` and all the fields within:
+
+```bash
+curl -X GET -H 'content-type: application/json' http://127.0.0.1:8080/configuration \
+    -d '{"Docs" : ["torii"]} ' -i
+```
+
+:::
+
+### Configuration
+
+::: warning
+
+TODO: Will change as part of the config RFC
+
+:::
+
+- **Protocol:** HTTP
+- **Method:** `POST`
+- **Endpoint:** `/configuration`
+- **Expects:**
+  - **Body:** One configuration option is currently supported: `LogLevel`. It is set to the log-level in uppercase.
+    ```json
+    {
+      "LogLevel": "WARN"
+    }
+    ```
+    Acceptable values are `TRACE`, `DEBUG`, `INFO`, `WARN`, `ERROR`.
+- **Responses:**
+  - `200 OK`: Log level has changed successfully. The confirmed new log level is returned in the body.
+  - `400 Bad Request`: request body malformed.
+  - `500 Internal Server Error`: Request body valid, but changing the log level failed (lock contention).
+
+### Health
+
+- **Protocol:** HTTP
+- **Method:** `GET`
+- **Endpoint:** `/health`
+
+Responses with `200 OK` and a current status of peer as a JSON string:
+
+```json
+"Healthy"
+```
+
+### Pending Transactions
+
+TODO: if it is for _internal use only_, remove this endpoint from the docs?
```

Comment:
I agree that if it's for internal use only, it shouldn't be here

If we don't want to lose this info, it might make sense to have a list of internal-use-only endpoints in the main repo